lib/versioning/loose/generic.ts

```diff
@@ -78,11 +78,19 @@ export const comparer = (
   }
 
   // we don't not have ranges, so versions has to be equal
-  function getSatisfyingVersion(versions: string[], range: string): string {
-    return versions.find((v) => equals(v, range)) || null;
-  }
-  function minSatisfyingVersion(versions: string[], range: string): string {
-    return versions.find((v) => equals(v, range)) || null;
+  function getSatisfyingVersion(
+    versions: string[],
+    range: string
+  ): string | null {
+    const result = versions.find((v) => equals(v, range));
+    return result?.length ? result : null;
```

Comment:
```suggestion
    return result ?? null;
```
Result of find is string or undefined.

So i think you only need to replace `||` with `??` in original code.